source/parser.hera

```diff
@@ -6633,6 +6633,23 @@ Reset
     Object.assign(module.config, parse.config)
     parse.config = module.config
 
+    // Hack to pass parser state to cache key
+    return {
+      type: "ParserMeta",
+      children: [],
+      getStateKey() {
+        const stateInt =
+          ((module.currentIndent.level % 256) << 8) |
+          (module.classImplicitCallForbidden << 7) |
+          (module.indentedApplicationForbidden << 6) |
+          (module.bracedApplicationForbidden << 5) |
+          (module.trailingMemberPropertyForbidden << 4) |
+          (module.newlineBinaryOpForbidden << 3)
```

Comment:
Other than current indentation level, we shouldn't really need shifting here, just ORing, because we're working with booleans. I confirmed that this is about 30% faster here: (please double check)

https://jsbench.me/mdllkhsbo3/1

Reply:
It only made around a ~1% difference for me. I think the cost of any bit operations is 100x less than the other stuff we're doing.

One possible optimization would be to try out different orders of the cache key parts to try and localize the partitions to where they are used the most in practice. (Splitting on pos, state, JSX, then rule maybe...)